### ISSUE TITLE
Does not Work With Parallel Steps in Pipeline

### ISSUE THREAD

**Issue by michael-hay** [jenkinsci/webhook-step-plugin#2]
I am trying to use this in some parallel steps in a pipeline like this:
```

params = ['1', '2']
series = [:]
for(def i=0; i < params.size(); i++) {
                    
                    def paramString = params[i]

                    series['Series' + (i+1)] = {
                        hook = registerWebhook()
                        echo "Waiting for POST to ${hook.getURL()}"
                        bat "curl -X POST http://user:pwd@localhost:8080/view/All/job/someJob/buildWithParameters?paramString=$paramString -v"
                        data = waitForWebhook hook
                        echo "Webhook received with data: ${data}"
                    }
 }
 parallel series

```
this will work if I only have a single series (nothing in parallel).  However as soon as I have more than one series in parallel I am not able to receive all the webhooks back to continue on.  This calling pipeline just hangs, even if I run the curl commands manually.

**Comment by cpitman** [comment]
I am trying to reproduce this, and the provided script is working exactly as expected:

```[Pipeline] parallel
[Pipeline] [Series1] { (Branch: Series1)
[Pipeline] [Series2] { (Branch: Series2)
[Pipeline] [Series1] registerWebhook
[Pipeline] [Series1] echo
[Series1] Waiting for POST to http://127.0.0.1:8080/webhook-step/664b4b0c-3d07-450a-b7c3-82371b152052
[Pipeline] [Series1] waitForWebhook
[Pipeline] [Series2] registerWebhook
[Pipeline] [Series2] echo
[Series2] Waiting for POST to http://127.0.0.1:8080/webhook-step/66c2f642-04ec-4735-9931-4429529970b9
[Pipeline] [Series2] waitForWebhook

[Pipeline] [Series1] echo
[Series1] Webhook received with data: test2
[Pipeline] [Series1] }

[Pipeline] [Series2] echo
[Series2] Webhook received with data: test2
[Pipeline] [Series2] }
[Pipeline] // parallel
[Pipeline] End of Pipeline
Finished: SUCCESS
```

I am using a slightly newer build, but the one defect fixed is #1, which shouldn't change behavior based on using parallel.

**Comment by cpitman** [comment]
I've released the updated version, in case it actually resolves this. Let me know if you continue seeing this issue!
